cpp/src/utilities/wrapper_types.hpp

```diff
@@ -152,28 +152,28 @@ template <typename T, gdf_dtype type_id>
 CUDA_HOST_DEVICE_CALLABLE
 wrapper<T,type_id> operator+(wrapper<T, type_id> const &lhs, wrapper<T, type_id> const &rhs)
 {
-  return wrapper<T, type_id>{lhs.value + rhs.value};
+  return wrapper<T, type_id>{ static_cast<T>(lhs.value + rhs.value) };
```

Comment:
Can you add a comment that this cast is required due to the implicit integral up-casting for types smaller than `int`?

Reply:
this is needed for int8 and smaller types since the OS casts to int32 for math operations , that causes a downcast error.